mod/caching/set/type_plus_right/wikirate_company/wikirate_topic.rb

```diff
@@ -45,13 +43,28 @@ def unique_metric_ids
   # pluck seems dumb here, but .all isn't working (returns *all card)
 end
 
+# @returns Array [[topic_card1, num_metrics], [topic_card2...]...] }
 def topics_by_metric_count
-  topic_count_hash =
-    item_cards(limit: 0).each_with_object({}) do |topic_card, count_hash|
-      count_hash[topic_card] = topic_card.metric_card.cached_count
-      count_hash
-    end
-  topic_count_hash.sort_by { |_card, count| count }.reverse
+  topic_metric_count_hash.sort_by { |_card, count| count }.reverse
```

Comment:
You didn't introduce it but `sort_by` is built for the case that key comparison is very expensive 
(e.g. every comparison creates a complicated object). In this case where comparison is simple the `sort` method is much faster.

Reply:
I don't think sort will cut it here.  We want to sort by the value.

```
 {a: 1, b: 10, c: 5 }.sort
=> [[:a, 1], [:b, 10], [:c, 5]]

{a: 1, b: 10, c: 5 }.sort_by { |_k, v| v }
=> [[:a, 1], [:c, 5], [:b, 10]]
```

Reply:
```{ a: 1, b: 10, c: 5 }.sort { |(k, v), (k1, v1)| v <=> v1 }`

Reply:
but it turns out for sorting by hash values `sort_by` is actually faster and looks better: